volatility3/framework/objects/__init__.py

```diff
@@ -141,7 +141,11 @@ def __getnewargs_ex__(self):
     @classmethod
     def _unmarshall(cls, context: interfaces.context.ContextInterface, data_format: DataFormatInfo,
                     object_info: interfaces.objects.ObjectInformation) -> TUnion[int, float, bool, bytes, str]:
-        data = context.layers.read(object_info.layer_name, object_info.offset, data_format.length)
+        # Don't try to lookup a 0 length data format, incase it's at an invalid offset.  Length 0 means b''
+        if data_format.length > 0:
```

Comment:
I would do something similar to this:
```
data = b''
if data_format.length > 0:
    data = ...
```

Reply:
Thanks, yep, I get what you meant now, I've added that, thanks.  5:)